open_rarity/resolver/testset_resolver.py

```diff
@@ -354,29 +359,34 @@ def augment_with_open_rarity_scores(
         )
 
 
-def extract_rank(tokens_to_score: dict[int, TokenRarity]) -> RankedTokens:
+def extract_rank(tokens_to_score: dict[str, TokenRarity]) -> RankedTokens:
     """Sorts dictionary by float score and extract rank according to the score
 
     Parameters
     ----------
-    token_id_to_scores : dict[int, TokenRarity]
+    token_id_to_scores : dict[str, TokenRarity]
         dictionary of token_id_to_scores with token_id to score mapping
 
     Returns
     -------
-    dict[int, RankScore]
+    dict[str, RankScore]
         dictionary of token to rank, score pair
     """
     ranked_tokens: list[TokenRarity] = RarityRanker.set_rarity_ranks(
-        token_rarities=tokens_to_score.values()
+        token_rarities=list(tokens_to_score.values())
     )
-    return {
-        int(token.token.token_identifier.token_id): (
+
+    result = {}
+    for token in ranked_tokens:
+        assert token.rank
+        # note: this is a bug, ignoring the mypy error for now
+        # token_identifier can be of type SolanaMintAddressTokenIdentifier
+        # which has no token_id
+        result[int(token.token.token_identifier.token_id)] = (  # type: ignore
```

Comment:
yeah we assert in caller thats its EVM token identifier but we do need to fix this